src/plugins/summaryWidget/src/Widget.js

```diff
@@ -0,0 +1,320 @@
+        this.domainObject.configuration.ruleOrder = this.domainObject.configuration.ruleOrder || ['default'];
+        this.domainObject.configuration.testDataConfig = this.domainObject.configuration.testDataConfig || [{
+            object: '',
+            key: '',
+            value: ''
+        }];
+
+        this.activeId = 'default';
+        this.rulesById = {};
+        this.domElement = $(widgetTemplate);
+        this.editing = false;
+        this.container = '';
+        this.ruleArea = $('#ruleArea', this.domElement);
+        this.testDataArea = $('.widget-test-data', this.domElement);
+        this.addRuleButton = $('#addRule', this.domElement);
+
+        this.conditionManager = new ConditionManager(this.domainObject, this.openmct);
+        this.testDataManager = new TestDataManager(this.domainObject, this.conditionManager, this.openmct);
+
+        this.show = this.show.bind(this);
+        this.destroy = this.destroy.bind(this);
+        this.addRule = this.addRule.bind(this);
+        this.refreshRules = this.refreshRules.bind(this);
+        this.duplicateRule = this.duplicateRule.bind(this);
+        this.updateWidget = this.updateWidget.bind(this);
+        this.executeRules = this.executeRules.bind(this);
+        this.reorder = this.reorder.bind(this);
+        this.onEdit = this.onEdit.bind(this);
+
+        var id = this.domainObject.identifier.key,
+            self = this,
+            oldDomainObject,
+            statusCapability;
+
+        openmct.$injector.get('objectService')
+            .getObjects([id])
+            .then(function (objs) {
+                oldDomainObject = objs[id];
+                statusCapability = oldDomainObject.getCapability('status');
+                statusCapability.listen(self.onEdit);
+                if (statusCapability.get('editing')) {
+                    self.onEdit(['editing']);
+                } else {
+                    self.onEdit([]);
+                }
+            });
+    }
+
+    /**
+     * Builds the Summary Widget's DOM, performs other necessary setup, and attaches
+     * this Summary Widget's view to the supplied container.
+     *
+     * @param {element} container The DOM element that will contain this Summary
+     *                            Widget's view.
+     */
+    Widget.prototype.show = function (container) {
+        this.container = container;
+        $(container).append(this.domElement);
+        $('.widget-test-data', this.domElement).append(this.testDataManager.getDOM());
+        this.widgetDnD = new WidgetDnD(this.domElement, this.domainObject.configuration.ruleOrder, this.rulesById);
+        this.initRule('default', 'Default');
+        this.refreshRules();
+        this.updateWidget();
+
+        this.addRuleButton.on('click', this.addRule);
+        this.conditionManager.on('receiveTelemetry', this.executeRules);
+        this.widgetDnD.on('drop', this.reorder);
+    };
+
+    Widget.prototype.destroy = function (container) {
+

```

Comment:
very important to implement `destroy`.  You need to remove any telemetry subscriptions, mutation or composition listeners, clean up event handlers, etc.